plugins/outputs/kinesis/kinesis.go

```diff
@@ -163,6 +163,21 @@ func writekinesis(k *KinesisOutput, r []*kinesis.PutRecordsRequestEntry) time.Du
 	return time.Since(start)
 }
 
+func (k *KinesisOutput) getPartitionKey(metric telegraf.Metric) string {
+	if k.RandomPartitionKey {
+		u := uuid.NewV4()
+		return u.String()
+	}
+	if k.PartitionKey != "" {
```

Comment:
This is the only part I'm not sure of, what does toml default too if a value isn't set?